test/test-tcp-rst.c

```diff
@@ -0,0 +1,84 @@
+#include "uv.h"
+#include "task.h"
+
+static uv_tcp_t tcp;
+static uv_connect_t connect_req;
+static uv_shutdown_t shutdown_req;
+static uv_buf_t qbuf;
+static int called_alloc_cb;
+static int called_connect_cb;
+static int called_shutdown_cb;
+static int called_close_cb;
+
+
+static void close_cb(uv_handle_t* handle) {
+  ASSERT(handle == (uv_handle_t*) &tcp);
+  called_close_cb++;
+}
+
+
+static void alloc_cb(uv_handle_t* handle, size_t size, uv_buf_t* buf) {
+  buf->base = malloc(size);
+  buf->len = size;
+  called_alloc_cb++;
+}
+
+
+static void read_cb(uv_stream_t* t, ssize_t nread, const uv_buf_t* buf) {
+  ASSERT_PTR_EQ((uv_tcp_t*)t, &tcp);
+  ASSERT_EQ(nread, UV_ECONNRESET);
+
+  uv_handle_type type = uv_guess_handle(t->io_watcher.fd);
```

Comment:
use `uv_fileno` here to get the `fd` field

```
D:\a\libuv\libuv\test\test-tcp-rst.c(31): error C2039: 'io_watcher': is not a member of 'uv_stream_s' [D:\a\libuv\libuv\build\uv_run_tests.vcxproj]
[464](https://github.com/libuv/libuv/runs/5290749931?check_suite_focus=true#step:4:464)
  D:\a\libuv\libuv\include\uv.h(504): note: see declaration of 'uv_stream_s'
[465](https://github.com/libuv/libuv/runs/5290749931?check_suite_focus=true#step:4:465)
D:\a\libuv\libuv\test\test-tcp-rst.c(31): error C2198: 'uv_guess_handle': too few arguments for call [D:\a\libuv\libuv\build\uv_run_tests.vcxproj]
[466](https://github.com/libuv/libuv/runs/5290749931?check_suite_focus=true#step:4:466)
```

Reply:
Done